packages/analytics-browser/src/config.ts

```diff
@@ -62,105 +53,145 @@ export class BrowserConfig extends Config implements IBrowserConfig {
   partnerId?: string;
   sessionTimeout: number;
   trackingOptions: TrackingOptions;
-  sessionManager: ISessionManager;
 
-  constructor(apiKey: string, userId?: string, options?: BrowserOptions) {
+  // NOTE: These protected properties are used to cache values from async storage
+  protected _deviceId?: string;
+  protected _lastEventTime?: number;
+  protected _optOut = false;
+  protected _sessionId?: number;
+  protected _userId?: string;
```

Comment:
These values are stored in persistent storage. Read and writes from/to persistent storage is asynchronous and would affect public facing APIs like `setUserId`, `getUserId`, etc. To keep public facing APIs synchronous, we cache these values in config. When reading, we return cached values. When writing we write to cache and storage at the same time.